src/wallet.rs

```diff
@@ -208,6 +208,112 @@ where
 		Ok(self.inner.lock().unwrap().get_balance()?)
 	}
 
+	pub(crate) fn send_to_address(
+		&self, address: &bitcoin::Address, amount_sats: u64,
+	) -> Result<Txid, Error> {
+		let confirmation_target = ConfirmationTarget::Normal;
+		let fee_rate = self.estimate_fee_rate(confirmation_target);
+
+		let tx = {
+			let locked_wallet = self.inner.lock().unwrap();
+			let mut tx_builder = locked_wallet.build_tx();
+
+			tx_builder
+				.add_recipient(address.script_pubkey(), amount_sats)
+				.fee_rate(fee_rate)
+				.enable_rbf();
+
+			let mut psbt = match tx_builder.finish() {
+				Ok((psbt, _)) => {
+					log_trace!(self.logger, "Created PSBT: {:?}", psbt);
+					psbt
+				}
+				Err(err) => {
+					log_error!(self.logger, "Failed to create transaction: {}", err);
+					return Err(err.into());
+				}
+			};
+
+			match locked_wallet.sign(&mut psbt, SignOptions::default()) {
+				Ok(finalized) => {
+					if !finalized {
+						return Err(Error::OnchainTxCreationFailed);
+					}
+				}
+				Err(err) => {
+					log_error!(self.logger, "Failed to create transaction: {}", err);
+					return Err(err.into());
+				}
+			}
+			psbt.extract_tx()
+		};
+
+		self.broadcast_transaction(&tx);
+
+		let txid = tx.txid();
+		log_info!(
+			self.logger,
+			"Created new transaction {} sending {}sats on-chain to address {}",
+			txid,
+			amount_sats,
+			address
+		);
+
+		Ok(txid)
+	}
+
+	pub(crate) fn drain_to_address(&self, address: &bitcoin::Address) -> Result<Txid, Error> {
```

Comment:
Could we dry this up by delegating to `send_to_address` or I suppose some common helper taking `Option<u64>` for the amount?